subprojects/launcher/src/integTest/groovy/org/gradle/launcher/CommandLineIntegrationSpec.groovy

```diff
@@ -239,4 +239,35 @@ class CommandLineIntegrationSpec extends AbstractIntegrationSpec {
         6                      | ['--parallel', '--max-workers=6', '-Dorg.gradle.workers.max=4']
         4                      | ['-Dorg.gradle.parallel=true', '--max-workers=4']
     }
+
+    @Unroll
+    def "Set log level using org.gradle.log.level"() {
+        def message = 'Expected message in the output'
+        buildFile << """
+            task assertLogging {
+                doLast {
+                    logger.${logLevel} '${message}'
+                    assert logger.${logLevel}Enabled
+                    if ('${nextLevel}') {
```

Comment:
It's the only way to test that the log level is actually set.  Otherwise if the logging was just ALWAYS `debug` all the tests would pass, because every level is enabled and everything would be outputted.  So we need to ensure that `quiet` means that both quiet is enabled, and warnings are not.